packages/mock/src/client.ts

```diff
@@ -743,12 +761,33 @@ export class MockFetchClient {
       headers: toIncomingHttpHeaders(options.headers),
     };
 
-    const result = await this.router.handleRequest(request, this.repo);
-    if (result.length === 1) {
-      return result[0];
-    } else {
-      return result[1];
+    const [outcome, resource] = await this.router.handleRequest(request, this.repo);
+    if (!resource) {
+      const issueDetails = outcome.issue[0]?.details;
+      const issueText = issueDetails?.text;
+      if (!issueText) {
+        return outcome;
+      }
+      // Special case for unknown search parameters and resource types
```

Comment:
We can discuss as a team.  I'm kinda philosophically opposed to having these kinds of extra layers of error logic.

I understand the utility of helping users who are fighting mysterious error messages.  That is a good and noble goal.  I'm just not sure this is the right place or the right way to do it.

This is brittle, because it depends on exact error message strings.  It also duplicates errors in multiple places.

I.e., I'd rather make it easier to automatically call `indexStructureDefinitionBundle` and `indexSearchParameterBundle` with a single medplum test setup utility or something.

Or, we could update all of the `throw new OperationOutcomeError` call sites to have short messages and longer detailed messages.

I'm open to improving the dev ex here, I'm just concerned about the sustainability and maintainability of this approach.